test/extensions/filters/http/decompressor/decompressor_filter_integration_test.cc

```diff
@@ -348,4 +348,79 @@ TEST_P(DecompressorIntegrationTest, DecompressAndBuffer) {
   EXPECT_EQ(1L, counter->value());
 }
 
+// Stop decompressing when output-buffer's size exceeds the number of
+// 'max_inflate_ratio*input-data-size'.
+TEST_P(DecompressorIntegrationTest, LimitMaxDecompressOutputSize) {
+  // Set max_inflate_ratio = 10.
+  initializeFilter(R"EOF(
+  name: envoy.filters.http.decompressor
+  typed_config:
+    "@type": type.googleapis.com/envoy.extensions.filters.http.decompressor.v3.Decompressor
+    decompressor_library:
+      name: testlib
+      typed_config:
+        "@type": "type.googleapis.com/envoy.extensions.compression.gzip.decompressor.v3.Gzip"
+        max_inflate_ratio: 10
+  )EOF");
+  auto encoder_decoder =
+      codec_client_->startRequest(Http::TestRequestHeaderMapImpl{{":method", "POST"},
+                                                                 {":scheme", "http"},
+                                                                 {":path", "/test/long/url"},
+                                                                 {"content-encoding", "gzip"},
+                                                                 {":authority", "host"}});
+
+  auto request_encoder = &encoder_decoder.first;
+  auto response = std::move(encoder_decoder.second);
+
+  // Compressed 20K zero bytes data.
+  constexpr uint8_t buffer[] = {0x1f, 0x8b, 0x08, 0x08, 0x8a, 0x51, 0xda, 0x62, 0x00, 0x03, 0x66,
+                                0x69, 0x6c, 0x65, 0x2e, 0x74, 0x78, 0x74, 0x00, 0xed, 0xc1, 0x31,
+                                0x01, 0x00, 0x00, 0x00, 0xc2, 0xa0, 0xf5, 0x4f, 0x6d, 0x0a, 0x3f,
+                                0xa0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x80, 0xb7,
+                                0x01, 0x60, 0x83, 0xbc, 0xe6, 0x00, 0x50, 0x00, 0x00};
+
+  // Note that the threshold is max_inflate_ratio*48 = 480 which is less than 20K.
+  Buffer::OwnedImpl data(buffer, 48);
```

Comment:
the buffer above has 4*11+9 => 53 bytes, yet we used 48 here?  Maybe using `sizeof(buffer)` might be more readable

Reply:
@KBaichoo  sry. I counted wrong carelessly. Thanks for your notice!